tests/integration/features/bootstrap/SharingContext.php

```diff
@@ -177,6 +186,10 @@ public function userSharesWithGroupWithOcs(string $user, string $path, string $s
 	 * @param TableNode|null $body
 	 */
 	public function userSharesWithRoom(string $user, string $path, string $room, TableNode $body = null) {
+		if (str_contains($path, '{{TOKEN}}')) {
+			$path = str_replace('{{TOKEN}}', $this->featureContext->getLastConversationToken(), $path);
+		}
```

Comment:
If I am not mistaken this does not have any effect (`{{TOKEN}}` is not included in `user XXX shares with room YYY` steps in the feature files), so it can be removed along with the code that was introduced to support it (like [`gatherContext`](https://github.com/nextcloud/spreed/pull/9326/commits/760f6b657ea4afeb134c2ad165cf1d13f7a9a272#diff-5a9151b805ab502cfdc791b951cf6a3d276f12e00023da65b62ea65f7b3b7bc1R61) or [`lastToken`](https://github.com/nextcloud/spreed/pull/9326/commits/760f6b657ea4afeb134c2ad165cf1d13f7a9a272#diff-8800f1da5bf5cabd2719341aee2113d530551a68869b93dc70edfa7384ccd7cdR100)).

In order to check that [the token in the notification matches](https://github.com/nextcloud/spreed/pull/9326/files#diff-3e8a31aeebbc4882795b94f01869b1803fe29c74de7a0c0df298a0e0f2a9b2e2R447-R477) rather than `{{TOKEN}}` something like `{{TOKEN:room1}}` (or whatever room identifier was used) could be used instead, replacing the room token in the original message by `'{{TOKEN:' . self::$tokenToIdentifier($notification['object_id']) . '}}'` in https://github.com/nextcloud/spreed/blob/760f6b657ea4afeb134c2ad165cf1d13f7a9a272/tests/integration/features/bootstrap/FeatureContext.php#L2797 (of course a regexp that matches against something like `{{TOKEN:[a-zA-Z0-9]*}}` instead of just `{{TOKEN}}` would be needed too in https://github.com/nextcloud/spreed/blob/760f6b657ea4afeb134c2ad165cf1d13f7a9a272/tests/integration/features/bootstrap/FeatureContext.php#L2796).

Reply:
It's from an earlier attempt, just forgot to remove from git add I think 🙈

Reply:
> In order to check that [the token in the notification matches](https://github.com/nextcloud/spreed/pull/9326/files#diff-3e8a31aeebbc4882795b94f01869b1803fe29c74de7a0c0df298a0e0f2a9b2e2R447-R477) rather than {{TOKEN}} something like {{TOKEN:room1}} (or whatever room identifier was used) could be used instead, replacing the room token in the original message by '{{TOKEN:' . self::$tokenToIdentifier($notification['object_id']) . '}}' in

That is all not new code, but yeah